searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/evaluation/StringValue.java

```diff
@@ -31,10 +35,15 @@ public StringValue(String value) {
     @Override
     public TensorType type() { return TensorType.empty; }
 
-    /** Returns the hashcode of this, to enable strings to be encoded (with reasonable safely) as doubles for optimization */
+    /**
+     * Returns the XXHash hashcode of this, to enable strings to be encoded (with reasonable safely)
+     * as doubles for optimization.
+     */
     @Override
     public double asDouble() {
-        return value.hashCode();
+        XXHash64 hasher = XXHashFactory.fastestInstance().hash64();
+        byte[] data = value.getBytes(StandardCharsets.UTF_8);
+        return Double.longBitsToDouble(hasher.hash(data, 0, data.length, 0));
```

Comment:
I guess we want to avoid the patterns that gives NaN as it does not compare well.
There is a suggestion in #24938 after reading the documentation for Double.longBitsToDouble

Reply:
Ugh, ok I'll do it

Reply:
I applied the same trick here as in the backend.